### DISCUSSION TITLE
Reasonable parameters for `SecondQuantizedMolecule()` call

### DISCUSSION THREAD

**Discussion by einarG-1qbit** [goodchemistryco/Tangelo#237] [Q&A] [answered]
Dear Tangelo devs,
I'm trying to get a qubit Hamiltonian for C2HF3 molecule. It's geomeetry is given below:
```
molecule = [
    ("C", (0.0, 0.430631, 0.000000)),
    ("C", (-0.704353, -0.686847, 0.0)),
    ("F", (1.314791, 0.500587, 0.0)),
    ("F",(-0.554822, 1.630861, 0.0)),
    ("F", (-0.092517, -1.882156, 0.0)),
    ("H",(-1.780948, -0.706341, 0.0))   
]
```
Could you recommend a reasonable set of parameters for the `SecondQuantizedMolecule()` call? Specifically, what are the most sensable  parameters for `spin`, `q` and `basis` in the call below?
`second_quant_mol = SecondQuantizedMolecule(molecule, q=0, spin=0, basis="sto-3g")`

If these parameters anyhow affect the size of the resulting qubit Hamiltonian, could you recommend some parameter values that will yield a 40-50 qubit Hamiltonian ?

The end goal is to get a 40-50 qubit quantum circuit by calling  `get_exponentiated_qubit_operator_circuit()`.

**Comment by alexfleury-sb** [comment, marked as answer]
Hi @einarG-1qbit, I took a look at your molecule and for this molecule, `spin=0` and `q=0` should do (it is a stable organic compound, therefore it shouldn't be an ion (where q=/=0) or an open-shell system (where spin=/=0). In the context of quantum simulation, those parameters are important for the initial state: it should have a non-zero overlap with the expected state at the end. If you are targeting electronic excited states, q and spin should change.

For the basis, `basis="sto-3g"` should also do for the first steps. In quantum chemistry, it is preferable to have an infinite basis set, but it is not tractable. If you are OK with it, I may suggest the "Choice of the Basis Set" in our [qchem_modelling_basics](https://github.com/goodchemistryco/Tangelo/blob/main/examples/qchem_modelling_basics.ipynb) notebook where you would find the effect of the basis set on the final state.

I tried with this code
```python
from tangelo import SecondQuantizedMolecule
from tangelo.toolboxes.operators import count_qubits
from tangelo.toolboxes.qubit_mappings.mapping_transform import fermion_to_qubit_mapping

xyz = [
    ("C", (0.0, 0.430631, 0.000000)),
    ("C", (-0.704353, -0.686847, 0.0)),
    ("F", (1.314791, 0.500587, 0.0)),
    ("F",(-0.554822, 1.630861, 0.0)),
    ("F", (-0.092517, -1.882156, 0.0)),
    ("H",(-1.780948, -0.706341, 0.0))
]

mol = SecondQuantizedMolecule(xyz, q=0, spin=0, basis="sto-3g", frozen_orbitals="frozen_core")
print(mol)

qubit_op = fermion_to_qubit_mapping(
    mol.fermionic_hamiltonian,
    mapping="JKMN",
    n_spinorbitals=mol.n_active_sos,
    n_electrons=mol.n_active_electrons,
    spin=mol.spin
)
print(count_qubits(qubit_op))
```
and got a qubit Hamiltonian of 42 qubits (thus as desired). 

The ` frozen_orbitals="frozen_core"` (default value for frozen_orbitals) argument is freezing the core molecular orbitals (MOs) in this molecule (a common approximation, quantum chemistry deals mainly with valence MOs). I also set `mapping="JKMN"`, as it is the most efficient mapping technique we have in tangelo regarding the length of the resulting Pauli words. In consequence, it should require less entangling gates per qubit term, in average.

**Reply by einarG-1qbit** [reply]
Thanks a lot Alexandre!
